packages/gluestick/src/renderer/render.js

```diff
@@ -29,7 +29,7 @@ type WrappersRequirements = {
   BodyWrapper: Object,
   entryWrapperConfig: Object,
   envVariables: any[],
-  entriesPlugins: { plugin: Function, meta: Object }[],
```

Comment:
I think you can improve this, and it'll help the plugin system a lot.

Flow makes Gluestick less readable, but it's not because of Flow. It's because of the large number of inlined types which get repeatedly redefined, as well as the types that are just _slightly_ different from existing types. In many cases, there's no reason for the slight differences.

In `src/types.js`, these are types for plugins. (I'm not sure what `Plugin` means in this context vs the others.)
```
export type Plugin = {
  name: string,
  meta: {
    [key: string]: any,
  },
  body: Function | null,
  options: Object,
};

export type ConfigPlugin = {
  name: string,
  meta: {
    [key: string]: any,
  },
  preOverwrites: {
    sharedWebpackConfig?: (config: WebpackConfig) => WebpackConfig,
  },
  postOverwrites: {
    gluestickConfig?: (config: GSConfig) => void,
    clientWebpackConfig?: (config: WebpackConfig) => WebpackConfig,
    serverWebpackConfig?: (config: WebpackConfig) => WebpackConfig,
    vendorDllWebpackConfig?: (config: WebpackConfig) => WebpackConfig,
  },
};

export type RuntimePlugin = {
  name: string,
  meta: {
    [key: string]: any,
  },
  body: {
    rootWrapper?: (component: Object) => Object,
  },
};

export type ServerPlugin = {
  name: string,
  meta: {
    [key: string]: any,
  },
  renderMethod: Function,
  hooks: GSHooks,
  logger: Logger,
};
```

Then there's a PluginRef from serverPlugins.js. I don't understand how or why the above type gets converted to this structure.
```
type PluginRef = {
  ref: Function,
  type: string,
  options?: Object,
};
```

and finally your inlined type, which seems to be different from all the above:
```
{ ref: { plugin: Function, meta: Object }, options?: Object }
```

I'm not opposed to having different types and signatures for different types of plugins (it prevents large numbers of optional types) as long as we keep these types intact as long as possible. Can you bring some sanity to these, and see if you can figure out how to avoid the addition of a new type specifically for `render.js`?